processing/src/main/java/io/druid/query/extraction/MapLookupExtractor.java

```diff
@@ -52,8 +53,8 @@ public MapLookupExtractor(
       @JsonProperty("isOneToOne") boolean isOneToOne
   )
   {
-    this.map = Preconditions.checkNotNull(map, "map");
     this.isOneToOne = isOneToOne;
+    this.map = Preconditions.checkNotNull(this.isOneToOne ? HashBiMap.create(map) : map, "map");
```

Comment:
It looks like `HashBiMap#inverse` creates a new object in a not thread safe manner, but the object is pretty light weight. Due to the nature of lookups this might cause a thundering of racy objects, which should be benign since the map is immutable, but annoying. Would it make sense to create the inverse map as a `Map<String, String>` eagerly, and `null` the object if not one-to-one, then in the unapply simply check for the eager inverse map being non null and do a `Collections.singletonList` on the result, if it exists?